C-Plus-Plus/cp/Leftmostnonrepeatingchar.cpp

```diff
@@ -0,0 +1,37 @@
+// Description: This program finds the leftmost non repeating character in a string.
+// For example: s = "aabcdb", output should be: 3 (index of c)
+#include<bits/stdc++.h>
+using namespace std;
+
+int leftmostnonrepeatingchar(string s){
+    int res = numeric_limits<int>::max();  //Initialised res to infinity
+    int fi[256];                           //An array to store first indices of string characters(size is 256 because there are 256 ascii characters)
```

Comment:
Instead of declaring an array with some size, try using inbuilt dynamic arrays, Hint - Vectors. The -1 initialization will be easier as well.
In this case you can use the following piece of code.
`std::vector<int> first_indices(256,-1);`

In vector declaration, the second parameter is the value to be initialized. Hope you got it now.

Reply:
Yes, I forgot about vector, I remember it now. Thanks!